manim/mobject/mobject.py

```diff
@@ -2618,6 +2642,21 @@ def construct(self):
                     circ.become(square)
                     self.wait(0.5)
         """
+
+        # changes the new mobject's height, width, center, or depth to the original's before transformation
+        # if both width and height are matched then the object is stretched to fit both
+        if match_height and match_width:
+            mobject.stretch_to_fit_height(self.height)
+            mobject.stretch_to_fit_width(self.width)
+        elif match_height:
+            mobject.match_height(self)
+        elif match_width:
+            mobject.match_width(self)
+        if match_center:
+            mobject.move_to(self.get_center())
+        if match_depth:
+            mobject.match_depth(self)
```

Comment:
Independent of the discussion above, these should be in a similar order, and in the current logic `match_depth` should also be queried with an `elif`.

```suggestion
        elif match_depth:
            mobject.match_depth(self)
        if match_center:
            mobject.move_to(self.get_center())
```